### PR TITLE
sched/pthread: Implement pthread_attr_[get|set]detachstate

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
specified here:
https://pubs.opengroup.org/onlinepubs/009695399/functions/pthread_attr_getdetachstate.html

## Impact
New API.

### CODE REVIEW

File: libs/libc/pthread/pthread_attr_setdetachstate.c [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ *
+ ****************************************************************************/
+
+int pthread_attr_setdetachstate(FAR pthread_attr_t *attr,
+                                int detachstate)
+{
+  if (!attr)
+    {
+      return EINVAL;
+    }
```
Comment (Ouss4) [inline libs/libc/pthread/pthread_attr_setdetachstate.c:53]: > The pthread_attr_setdetachstate() function shall fail if:
[EINVAL]
      The value of detachstate was not valid 

We need to also check for `detachstate` to be valid, either: PTHREAD_CREATE_DETACHED or PTHREAD_CREATE_JOINABLE.

Comment (Ouss4) [reply libs/libc/pthread/pthread_attr_setdetachstate.c:53]: BTW, just a note: [Issue 7](https://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_attr_getdetachstate.html#tag_16_404) specifies that the return is undefined if `attr` is not valid, which is different from issue 6 (the link in the description) where it explicitly specifies that EINVAL should be returned.

File: include/pthread.h [hunk 2]
```diff
@@ -241,7 +246,7 @@ struct pthread_attr_s
   uint8_t priority;            /* Priority of the pthread */
   uint8_t policy;              /* Pthread scheduler policy */
   uint8_t inheritsched;        /* Inherit parent priority/policy? */
-
+  uint8_t detachstate;         /* Initilize to the detach state */
```
[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply include/pthread.h]: Done.

File: libs/libc/pthread/pthread_attr_setdetachstate.c (hunk 1 above)
Comment (xiaoxiang781216) [reply libs/libc/pthread/pthread_attr_setdetachstate.c]: Done.

**Review by Ouss4 — APPROVED** [review]

**Pull request merged**